cmd/test_runner_test.go

```diff
@@ -182,56 +183,103 @@ func (run *TestRunner) MakeTmpDir(t *testing.T, pattern string) string {
 
 // ExecCommand executes the given command string
 func (run *TestRunner) ExecCommand(cmdText string) error {
-	run.CmdR = run.CreateCommandRunner(run.Context)
-	return executeCommand(run.CmdR, cmdText)
+	var doneCh chan struct{}
+	ctx, cancel := context.WithCancel(run.Context)
+	run.CmdR, doneCh = run.CreateCommandRunner(ctx)
+	if err := executeCommand(run.CmdR, cmdText); err != nil {
+		cancel()
+		return err
+	}
+	cancel()
+	if doneCh != nil {
+		<-doneCh
+	}
+	return nil
 }
 
 // ExecCommandWithStdin executes the given command string with the string as stdin content
 func (run *TestRunner) ExecCommandWithStdin(ctx context.Context, cmdText, stdinText string) error {
 	setNoColor(true)
-	cmd := NewQriCommand(ctx, run.pathFactory, run.RepoRoot.TestCrypto, run.Streams)
+	ctx, cancel := context.WithCancel(ctx)
+	cmd, doneCh := NewQriCommand(ctx, run.pathFactory, run.RepoRoot.TestCrypto, run.Streams)
 	cmd.SetOutput(run.OutStream)
 	run.CmdR = cmd
-	return executeCommand(run.CmdR, cmdText)
+	if err := executeCommand(run.CmdR, cmdText); err != nil {
+		cancel()
+		return err
+	}
+	cancel()
+	if doneCh != nil {
+		<-doneCh
+	}
+	return nil
 }
 
 // ExecCommandCombinedOutErr executes the command with a combined stdout and stderr stream
 func (run *TestRunner) ExecCommandCombinedOutErr(cmdText string) error {
-	run.CmdR = run.CreateCommandRunnerCombinedOutErr(run.Context)
-	return executeCommand(run.CmdR, cmdText)
+	ctx, cancel := context.WithCancel(run.Context)
+	var doneCh chan struct{}
+	run.CmdR, doneCh = run.CreateCommandRunnerCombinedOutErr(ctx)
+	if err := executeCommand(run.CmdR, cmdText); err != nil {
+		cancel()
+		return err
+	}
+	cancel()
+	if doneCh != nil {
+		<-doneCh
+	}
+	return nil
 }
 
 // ExecCommandWithContext executes the given command with a context
 func (run *TestRunner) ExecCommandWithContext(ctx context.Context, cmdText string) error {
```

Comment:
Do we even need this as it all runs with the `run.Context`. Now it's pretty much the same as `ExecCommand` or should we work off the provided context.